File: src/main/java/me/sargunvohra/leveluphp/Capabilities.java
package me.sargunvohra.leveluphp;

import lombok.experimental.UtilityClass;
import me.sargunvohra.leveluphp.data.ILevelHandler;
import net.minecraftforge.common.capabilities.Capability;
import net.minecraftforge.common.capabilities.CapabilityInject;

/** Utility class to contain capabilities owned by this mod. */
@UtilityClass
public class Capabilities {
  @CapabilityInject(ILevelHandler.class)
  public static Capability<ILevelHandler> LEVEL_HANDLER = null;
}

File: src/main/kotlin/me/sargunvohra/leveluphp/Capabilities.kt
package me.sargunvohra.leveluphp

import net.minecraftforge.common.capabilities.Capability
import net.minecraftforge.common.capabilities.CapabilityInject

object Capabilities {

    @JvmStatic
    @CapabilityInject(LuhpData::class)
    lateinit var LUHP_DATA: Capability<LuhpData>
}
